nextcord/interactions.py

```diff
@@ -1252,3 +1257,43 @@ class InteractionMessage(_InteractionMessageMixin, Message):
     """
 
     pass
+
+
+class MessageInteraction:
+    """Represents a message's interaction data, regardless of application.
+
+    A message's interaction data is a property of a message when the message
+    is a response to an interaction from any bot.
+    .. versionadded:: 2.0
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    name: :class:`str`
+        The application ID that the interaction was for.
+    user: :class:`User`
+        The user that sent the interaction.
+    data: :class:`dict`
+        The raw data from the interaction.
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "_state",
+        "data",
+        "id",
+        "type",
+        "name",
+        "user",
+    )
+
+    def __init__(self, *, data: MessageInteractionPayload, state: ConnectionState):
+        self._state: ConnectionState = state
+
+        self.data: MessageInteractionPayload = data
+        self.id: int = int(data["id"])
+        self.type: InteractionType = data["type"]
+        self.name: str = data["name"]
+        self.user: User = self._state.store_user(data["user"])
```

Comment:
```suggestion
        self.user: User = User(data["user"])
```
it's already stored, no need to re-store it